src/colorizer/CanvasOverlay.ts

```diff
@@ -445,17 +463,22 @@ export default class CanvasOverlay implements IRenderCanvas {
    * @param doesInnerCanvasNeedRender Whether the inner canvas needs to be
    * re-rendered. True by default.
    */
-  render(doesInnerCanvasNeedRender: boolean = true): void {
+  render(options?: OverlayRenderOptions): void {
     // Expand size by header + footer, if rendering:
-    const headerRenderer = this.getHeaderRenderer(this.isHeaderVisibleOnExport && this.isExporting);
-    const footerRenderer = this.getFooterRenderer(this.isFooterVisibleOnExport && this.isExporting);
+    const headerRenderer = this.getHeaderRenderer(this.exportOptions.showHeader && this.isExporting);
+    const footerRenderer = this.getFooterRenderer(this.exportOptions.showFooter && this.isExporting);
     this.headerSize = headerRenderer.sizePx;
     this.footerSize = footerRenderer.sizePx;
 
     // Update canvas resolution + size.
     const devicePixelRatio = getPixelRatio();
-    const baseCanvasWidthPx = toEven(this.innerCanvasSize.x);
-    const baseCanvasHeightPx = toEven(this.innerCanvasSize.y + this.headerSize.y + this.footerSize.y);
+
+    let baseCanvasWidthPx = this.innerCanvasSize.x;
+    let baseCanvasHeightPx = this.innerCanvasSize.y + this.headerSize.y + this.footerSize.y;
+    if (this.isExporting && this.exportOptions.enforceEven) {
+      baseCanvasWidthPx = toEven(baseCanvasWidthPx * devicePixelRatio) / devicePixelRatio;
+      baseCanvasHeightPx = toEven(baseCanvasHeightPx * devicePixelRatio) / devicePixelRatio;
```

Comment:
I try to explain this elsewhere, but internally the canvas's internal resolution is determined by `baseCanvasWidthPx * devicePixelRatio`. This step here ensures that the final resulting canvas resolution will have even dimensions if it's a requirement.